refactor: 피드 아이템 상세 기능 리팩터링

s9hn opened the pull request:
## 📌𝘐𝘴𝘴𝘶𝘦𝘴
- closed #34 

## 📎𝘞𝘰𝘳𝘬 𝘋𝘦𝘴𝘤𝘳𝘪𝘱𝘵𝘪𝘰𝘯
- [리뷰요청 커밋 기록](https://github.com/Team-WSS/WSS-Android/pull/117/files/6b2ad3942cf2c0ec30ccae555f0556feb5d41047..454baa5f67bacee3708837a81c9205d21d686142)
- 각 피드 아이템에 대한 상세 기능 구현 완료했습니다.
    - 수정됨 분기처리
    - 스포일러 피드 분기처리
    - 날짜 표기
    - 더보기 메뉴 리팩터링
    - 좋아요 기능 리팩터링

## 📷𝘚𝘤𝘳𝘦𝘦𝘯𝘴𝘩𝘰𝘵
[123.webm](https://github.com/user-attachments/assets/6ae64847-3729-4cd7-a11e-cf1d5ec125df)


## 💬𝘛𝘰 𝘙𝘦𝘷𝘪𝘦𝘸𝘦𝘳𝘴

s9hn commented on app/src/main/java/com/teamwss/websoso/ui/novelDetail/NovelDetailViewModel.kt:
```diff
@@ -60,6 +79,33 @@ class NovelDetailViewModel : ViewModel() {
+            ReadStatus.WISH -> "읽고 싶음"
+        }
+        _readStatusImage.value = when (status) {
+            ReadStatus.FINISH -> R.drawable.ic_status_finish
```
drawable 파일들은 view일까요 아닐까요?

m6z1 replied:
뷰 아니지 않을까요?! 이미지도 상태이니?????????? (_ _

s9hn replied:
헉 이미지가 상태인가요.........? R.drawable의 R이 무엇의 약자인가요..

m6z1 replied:
..resorce

s9hn replied:
이미지, 폰트, 색상 등은 뷰인가요? 뷰상태인가요?

m6z1 replied:
view

s9hn committed:
refactor: optimisticUI 및 디바운스 구현, 적용

- SingleEventHandler Debounce, throttle 함수 구현 및 네이밍 수정
- Fragment, ScrollListener 적용

s9hn committed:
refactor: optimisticUI 리팩터링

- 좋아요를 누를때마다 uistate를 생성하고 관찰 및 업데이트가 아닌 해당 아이템의 ui 즉각 업데이트 로직 추가

s9hn committed:
feat: 좋아요 API 연결 및 분기처리 구현

s9hn committed:
refactor: 좋아요 수 업데이트 구현 및 불필요한 리스트 갱신 애니메이션 삭제

s9hn committed:
refactor: 유저 차단 함수 삭제

s9hn committed:
refactor: 좋아요 해제 시 음수 값 예외처리

s9hn committed:
Merge pull request #115 from Team-WSS/feat/35

refactor: 좋아요 기능 리팩터링

s9hn committed:
Merge branch 'feat/34' of https://github.com/Team-WSS/WSS-Android into feat/48

s9hn committed:
refactor: FeedUiModel, isMyFeed 프로퍼티 추가

s9hn committed:
refactor: isMyFeed에 따른 매뉴 UI 및 구성 변경 분기처리 구현

s9hn committed:
feat: 팝업 메뉴에 따른 다이얼로그 UI 구현

s9hn committed:
feat: 팝업 매뉴 선택 시 해당 다이얼로그 생성 분기로직 구현

s9hn committed:
feat: 삭제 다이얼로그 뷰 로직 구현 및 데이터바인딩 적용

s9hn committed:
feat: 신고 다이얼로그 뷰 로직 구현 및 데이터바인딩 적용

s9hn committed:
feat: 베이스 DialogFragment 구현

s9hn committed:
refactor: 다이얼로그 배경 흐림 처리 구현

s9hn committed:
feat: 다이얼로그 버튼 클릭 리스너 구현

s9hn committed:
feat: 완료 다이얼로그 구현

s9hn committed:
refactor: crossinline -> noinline 수정

s9hn committed:
feat: 제거된 피드 ui업데이트 및 서버통신 로직 구현

s9hn committed:
feat: 스포일된 피드 ui업데이트 및 서버통신 로직 구현

s9hn committed:
feat: 부적절한 표현 피드 ui업데이트 및 서버통신 로직 구현

s9hn committed:
refactor: 팝업매뉴 선택 시 매뉴 즉시 닫히도록 수정

s9hn committed:
refactor: 다이얼로그 확인 시 매뉴 즉시 닫히도록 수정

s9hn committed:
refactor: 메서드 네이밍 수정

s9hn committed:
refactor: 메서드 위치 수정 및 임포트 제거

s9hn committed:
refactor: ui업데이트 로직 수정

s9hn committed:
refactor: defaultFeedRepository 네이밍 수정

s9hn committed:
feat: 레포지토리 모듈 및 서비스 모듈 구현

s9hn committed:
refactor: 네트워크 모듈 코드 리팩터링

s9hn committed:
feat: 스포일러 피드 숨김처리 구현

s9hn committed:
feat: 피드 날짜 및 수정됨 구현

s9hn committed:
refactor: bindMenuByIsMyFeed 분기문 함수 분리

s9hn committed:
Merge pull request #116 from Team-WSS/feat/48

feat: 더보기 기능 추가 구현

s9hn committed:
Merge branch 'feat/34' of https://github.com/Team-WSS/WSS-Android into feat/34

s9hn committed:
refactor: xml string 추출

librarywon commented on app/src/main/java/com/teamwss/websoso/ui/feed/FeedFragment.kt:
```diff
@@ -104,47 +104,56 @@ class FeedFragment : BindingFragment<FragmentFeedBinding>(R.layout.fragment_feed
-                    tvFeedPopupFirstItem.isSelected = false
-                    tvFeedPopupSecondItem.isSelected = false
-                }
+                true -> setupMyFeed(feedId, popup)
+                false -> setupNotMyFeed(feedId, popup)
             }
         }
     }
 
+    private fun MenuFeedPopupBinding.setupNotMyFeed(
+        feedId: Long,
+        popup: PopupWindow,
+    ) {
+        onFirstItemClick = {
+            showDialog<DialogReportPopupMenuBinding>(
+                title = getString(R.string.report_popup_menu_spoiling_feed),
+                event = { feedViewModel.updateReportedSpoilerFeed(feedId) },
+            )
+            popup.dismiss()
+        }
+        onSecondItemClick = {
+            showDialog<DialogReportPopupMenuBinding>(
+                title = getString(R.string.report_popup_menu_impertinence_feed),
+                event = { feedViewModel.updateReportedImpertinenceFeed(feedId) },
+            )
+            popup.dismiss()
+        }
+        menuContentTitle =
+            getString(R.string.feed_popup_menu_content_report_isNotMyFeed).split(",")
+        tvFeedPopupFirstItem.isSelected = false
+        tvFeedPopupSecondItem.isSelected = false
+    }
```
a: 히히 이쁘다

m6z1 reviewed (approved):
SGTM ~

librarywon reviewed (approved):
고생하셨습니다 짧아서 꼼꼼히 봤는데 변동사항이 없어서 어프드립니다!
